### PR TITLE
UI: swap `react-sizeme` for `react-resize-detector`

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/20140

## What I did

Migrated us to the proposed package.

### CODE REVIEW

File: code/addons/a11y/src/components/A11YPanel.test.tsx [hunk 1]
```diff
@@ -8,6 +8,9 @@ import { A11YPanel } from './A11YPanel';
 import { EVENTS } from '../constants';
 
 jest.mock('@storybook/manager-api');
+
+global.ResizeObserver = require('resize-observer-polyfill');
```
Comment (IanVS) [inline code/addons/a11y/src/components/A11YPanel.test.tsx:12]: Why is a polyfill required?  ResizeObserver seems well-supported, given that we support Chrome 100+ now.  https://caniuse.com/resizeobserver

File: code/addons/jest/src/components/Result.tsx [hunk 2]
```diff
@@ -86,5 +86,3 @@ export function Result(props: ResultProps) {
     </Fragment>
   );
 }
-
-export default Result;
```
Comment (IanVS) [inline code/addons/jest/src/components/Result.tsx:90]: This isn't a component that users might import directly, right?

**Review by IanVS — APPROVED** [review]
I had a couple of questions, but looks good otherwise.

File: code/addons/a11y/src/components/A11YPanel.test.tsx (hunk 1 above)
Comment (joshwooding) [reply code/addons/a11y/src/components/A11YPanel.test.tsx:12]: I guess this test is run in jest and I don't believe jsdom supports resizeobserver

Comment (ndelangen) [reply code/addons/a11y/src/components/A11YPanel.test.tsx:12]: bingo

File: code/addons/jest/src/components/Result.tsx (hunk 2 above)
Comment (ndelangen) [reply code/addons/jest/src/components/Result.tsx:90]: can't, it's bundled

**Review by joshwooding — APPROVED** [review]

**Pull request merged**